apps/student-app/src/app/student-app.tsx

```diff
@@ -0,0 +1,379 @@
+    } catch (err) {}
+  }
+
+  // Handles course code submission
+  async function handleSubmitCourse(e: React.FormEvent<HTMLFormElement>): Promise<void> {
+    e.preventDefault();
+    try {
+      // GET on course GUID
+      // STUB getting the data
+      setData(await submitCourse(course_code));
+      populate_word_list_items(); //Create the word lists List Items elements
+      setError('null');
+      setScreen(StudentState.word_list_state);
+    } catch (err) {
+      setError('error-course');
+    }
+  }
+
+  // Handles setting the word list to chosen list item ID
+  async function handleSetWordList(
+    id: string
+  ): Promise<void> {
+    try {
+      ;
+      setSelectedWordList(data?.word_lists.find((wrd_lst: WordList) => {
+        return wrd_lst.id === id;
+      }) ?? null)
+      setError('null');
+    } catch (err) {
+      setError('error-word-list');
+    }
+  }
+
+  // Handles text entry into Course Code Text field
+  function handleCourseChange(e: {
+    target: { value: React.SetStateAction<string> };
+  }): void {
+    setCourse(e.target.value);
+  }
+
+
+  return (
+    <>
+      <div id="splash-screen"
+        className={screen === StudentState.splash_state ? 'content' : 'hidden-content'}
+      >
+        <h1>Vocab Game</h1>
+        <div>Splash Screen</div>
+        <div>tap screen to start</div>
+        <button onClick={handleToCourse} id="start-btn" disabled={false}>
+          Start
+        </button>
+      </div>
+
+      <div id="course-selection"
+        className={screen === StudentState.course_state ? 'content' : 'hidden-content'}
+      >
+        <h1>Vocab Game</h1>
+        <div id="course-select">
+          <form id="course-select-form" onSubmit={handleSubmitCourse}>
+            {error === 'error-course' ? (
+              <span id="errorCourse" className="error">
+                Failed to load course guid.
+              </span>
+            ) : null}
+            <label>Enter Course Code: </label>
+            <input
+              type="text"
+              id="course-guid"
+              name="course-guid"
+              placeholder="qwerty"
+              value={course_code}
+              onChange={handleCourseChange}
+            />
+            <br></br>
+            <button type="submit" id="submit-course-code" disabled={false}>
+              Submit
+            </button>
+          </form>
+          <button id="back-to-splash-btn" onClick={handleBackToSplash}>
+            Back
+          </button>
+        </div>
+      </div>
+
+      <div
+        id="word-list-selection"
+        className={screen === StudentState.word_list_state ? 'content' : 'hidden-content'}
+      >
+        <h1>Vocab Game</h1>
+        {error === 'error-course' ? (
+          <span id="errorCourse" className="error">
+            Failed to load course guid.
+          </span>
+        ) : null}
+        <div id="word-list-select">
+          <form id="course-select-form" onSubmit={handleSubmitCourse}>
```

Comment:
the `onSubmit` isn't really a thing here, no submit button is necessary. A user advances by selecting a word-list

Reply:
alright, done